src/fides/api/service/privacy_request/request_runner_service.py

```diff
@@ -137,99 +195,134 @@ def get_manual_webhook_erasure_inputs(
         logger.info(exc)
         privacy_request.status = PrivacyRequestStatus.requires_input
         privacy_request.save(db)
-        return ManualWebhookResults(manual_data=manual_inputs, proceed=False)
+        return ManualWebhookResults(
+            manual_data_for_upload=manual_inputs,
+            manual_data_for_storage=manual_inputs,
+            proceed=False,
+        )
 
-    return ManualWebhookResults(manual_data=manual_inputs, proceed=True)
+    return ManualWebhookResults(
+        manual_data_for_upload=manual_inputs,
+        manual_data_for_storage=manual_inputs,
+        proceed=True,
+    )
 
 
-def run_webhooks_and_report_status(
-    db: Session,
+@log_context(capture_args={"privacy_request_id": LoggerContextKeys.privacy_request_id})
+def upload_access_results(
+    session: Session,
+    policy: Policy,
+    rule: Rule,
+    results_to_upload: dict[str, list[dict[str, Optional[Any]]]],
+    dataset_graph: DatasetGraph,
     privacy_request: PrivacyRequest,
-    webhook_cls: WebhookTypes,
-    after_webhook_id: Optional[str] = None,
-) -> bool:
-    """
-    Runs a series of webhooks either pre- or post- privacy request execution, if any are configured.
-    Updates privacy request status if execution is paused/errored.
-    Returns True if execution should proceed.
-    """
-    webhooks = db.query(webhook_cls).filter_by(policy_id=privacy_request.policy.id)  # type: ignore
+    upload_attachments: Iterator[AttachmentData],
+) -> list[str]:
+    """Upload results for a single rule and return download URLs and modified results."""
+    start_time = time.time()
+    download_urls: list[str] = []
+    storage_destination = rule.get_storage_destination(session)
 
-    if after_webhook_id:
-        # Only run webhooks configured to run after this Pre-Execution webhook
-        pre_webhook = PolicyPreWebhook.get(db=db, object_id=after_webhook_id)
-        webhooks = webhooks.filter(  # type: ignore[call-arg]
-            webhook_cls.order > pre_webhook.order,  # type: ignore[union-attr]
+    if upload_attachments:
+        results_to_upload["attachments"] = upload_attachments  # type: ignore[assignment]
```

Comment:
`upload_attachments` is an iterator so we'll run into issues assigning this to `results_to_upload["attachments"]` which expects a dict. I see you flagged this with `ignore[assignment]` but can you double check to make sure this is safe? I see we have `test_privacy_request_with_attachments` but because the `upload()` is mocked, we won't see if there are any issues

Reply:
Maybe something like this?
```suggestion
        results_to_upload["attachments"] = [attachment.to_upload_dict() for attachment in upload_attachments]
```

Reply:
Oh thats my bad, its already run through that function before this point. I had the wrong type hint there. Updated to `list[dict]`